sphinx/source/developers/consensus.rst

```diff
@@ -0,0 +1,41 @@
+Consensus Protocols
+===================
+
+In order to guarantee system reliability changes to the key-value store must be agreed by a number of nodes before being applied. The number of nodes that need to agree on a change before it is applied depends on the consensus protocol being run.
+
+The default consensus protocol for CCF is a version of Raft which provides Crash Fault Tolerance.
+
+We are currently working on providing PBFT as an additional consensus option. PBFT provides Byzantine Fault Tolerance.
+
+.. warning:: Currently CCF with PBFT is in development and should not be used in a production environment.
+
+Raft Consensus Protocol
```

Comment:
Could we merge this with https://microsoft.github.io/CCF/developers/ledger.html#ledger-replication ?

Reply:
In the same page? Why not give it its own page, it's quite a bit of text and a separate entity.

Reply:
I actually think that the `Ledger Replication` should go completely as it is incomplete and too detailed for this docs.

Reply:
I don't think it hurts to have the Ledger page under the developer docs. We can clean it up but it will probably be useful when we need to write about ledger auditability and verifiability.